substrate-archive-backend/src/read_only_backend.rs

```diff
@@ -44,23 +42,25 @@ use sp_runtime::{
     Justification,
 };
 use std::{convert::TryInto, sync::Arc};
+use substrate_archive_common::database::ReadOnlyDatabaseTrait;
+use substrate_archive_common::error::Result;
 
 pub struct ReadOnlyBackend<Block: BlockT> {
-    db: Arc<ReadOnlyDatabase>,
+    db: Arc<dyn ReadOnlyDatabaseTrait>,
```

Comment:
Just to state the obvious here: are we concerned about the binary size, or is there another reason to use dyn dispatch instead of static generics? Is the database access so infrequent we don't care about jumping through vtables for it?

Reply:
TBH it didn't cross my mind to use static generics when designing the ReadOnlyBackend the first time around, and it might be a worthy endeavour so I created #115. Binary size shouldn't be much of an issue since we aren't running on very constrained devices but rather servers that should have more than enough storage space.

Reply:
Done, transitioned to static generics.